### ISSUE TITLE
Periodic Task will not enqueue on startup, can't remove it or start it using Grunt

### ISSUE THREAD

**Issue by awinder** [evantahler/actionhero#654]
I have a periodic task that seems to get stuck on unclean shutdowns (i.e., killing the actionhero process and not allowing it to shut down cleanly).  When I bring the service back up, it never gets re-enqueued.  I saw the `grunt` helpers that seemed like they would be able to clear this problem up, but I'm getting the following output attempting to use them:

```
# grunt stopPeriodicTask:cacheToken
Running "stopPeriodicTask:cacheToken" (stopPeriodicTask) task
removed 0 instances of cacheToken

Done, without errors.
# grunt enqueuePeriodicTask:cacheToken
Running "enqueuePeriodicTask:cacheToken" (enqueuePeriodicTask) task
cacheToken not enqueued
```

Not really sure what to do other than to rename the periodic task, but this has happened a number of times so I'm kinda worried that this might be a not-so-rare occurance

**Comment by evantahler** [comment]
**tl;dr**: it's important to actually shut down workers properly.

---

In the situation you describe, this is actually the expected behavior, and here's why:

- When you `kill 9` and actionhero worker, the redis keys which indicated that the periodic tasks was claimed and being worked on are never removed:

![screen shot 2015-05-13 at 2 24 01 pm](https://cloud.githubusercontent.com/assets/303226/7611337/bd13b3ca-f97b-11e4-99f5-0aff548f033d.png)

The periodic tasks use 3 [resque plugins](https://github.com/evantahler/actionhero/blob/master/initializers/tasks.js#L44-L48) `jobLock`, `queueLock`, and `workerLock`, which in turn, prevent more than one instance of that job ever being enqueued, enqueued in the delayed queue, or being worked on... which it totally what you want.  

Now, with those keys still being in redis, when actionhero starts and checks on the delayed jobs which should be enqueued, it sees an (old) worker already working on your `cacheToken` job, so it thinks all is well and moves on. These resque plugins do have [timeouts](https://github.com/taskrabbit/node-resque/blob/master/lib/plugins/jobLock.js#L68-L75), but waiting that long is probably not a good idea.

Resque is designed not have a heartbeat or any real worker status mechanism, because it assumes single threaded applications/languages like node.  If your task is CPU-intensive, the event loop will be blocked, so it can't "ping" redis every so often with a "I'm still working" message.  It's up to you to handle failures/errors with your periodic tasks.

We recently added 2 methods to node-resque which are applicable here: `queue.cleanOldWorkers(age, callback)` and `queue.forceCleanWorker(workerName, callback)`.  I'm opting *not* to use these methods within actionhero by default, as they are dangerous (you can read the caveats [here](https://github.com/taskrabbit/node-resque#failed-worker-managment) and the related PR for actionhero [here](https://github.com/evantahler/actionhero/pull/655))

So the action for you here is to make an initializer that runs at boot to check for these dead workers using your own heuristics (Can you tell they are dead from the start time of the job?  Can you tell they are dead from the host name?).  Once you find those dead workers, you can use `queue.forceCleanWorker(workerName, callback)` (once PR #655 gets merged)

**Comment by evantahler** [comment]
This issue is closed in favor of #655
